tried adding this to the update platform and got

```
RecursionError: maximum recursion depth exceeded
ERROR:homeassistant.components.update:Error while setting up test platform for update
Traceback (most recent call last):
  File "/Users/bdraco/home-assistant/homeassistant/helpers/entity_platform.py", line 368, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/Users/bdraco/home-assistant/homeassistant/helpers/entity_platform.py", line 507, in async_add_entities
    await asyncio.gather(*tasks)
  File "/Users/bdraco/home-assistant/homeassistant/helpers/entity_platform.py", line 671, in _async_add_entity
    original_device_class=entity.device_class,
                          ^^^^^^^^^^^^^^^^^^^
  File "/Users/bdraco/home-assistant/tests/common.py", line 1149, in device_class
    return self._handle("device_class")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bdraco/home-assistant/tests/common.py", line 1225, in _handle
    return getattr(super(), attr)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bdraco/home-assistant/homeassistant/backports/functools.py", line 70, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/Users/bdraco/home-assistant/homeassistant/components/update/__init__.py", line 246, in device_class
    if hasattr(self, "_attr_device_class"):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bdraco/home-assistant/homeassistant/helpers/entity.py", line 306, in _getter
    return getattr(o, private_attribute_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bdraco/home-assistant/homeassistant/helpers/entity.py", line 306, in _getter
    return getattr(o, private_attribute_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bdraco/home-assistant/homeassistant/helpers/entity.py", line 306, in _getter
    return getattr(o, private_attribute_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  [Previous line repeated 734 more times]
RecursionError: maximum recursion depth exceeded
```

--- a/homeassistant/helpers/entity.py
+++ b/homeassistant/helpers/entity.py
@@ -323,26 +323,33 @@ class CachedProperties(type):
         def move_attr(cls: CachedProperties, property_name: str) -> None:
             attr_name = "_attr_" + property_name
             private_attr_name = "__attr_" + property_name
-            if hasattr(cls, attr_name):
+            if attr_name in cls.__dict__:
                 setattr(cls, private_attr_name, getattr(cls, attr_name))
                 annotations = cls.__annotations__
                 if attr_name in annotations:
                     annotations[private_attr_name] = annotations.pop(attr_name)
-            setattr(cls, attr_name, make_property(property_name))
+            type.__setattr__(cls, attr_name, make_property(property_name))
 
         cached_properties: set[str] = namespace["_CachedProperties__cached_properties"]
+        moved_attrs: set[str] = set()
         for property_name in cached_properties:
+            if property_name in moved_attrs:
+                continue
             move_attr(cls, property_name)
+            moved_attrs.add(property_name)
 
         for parent in cls.__mro__[:0:-1]:
             if not hasattr(parent, "_CachedProperties__cached_properties"):
                 continue
             cached_properties = getattr(parent, "_CachedProperties__cached_properties")
             for property_name in cached_properties:
+                if property_name in moved_attrs:
+                    continue
                 attr_name = "_attr_" + property_name
                 if (attr_name) not in cls.__dict__:
                     continue
                 move_attr(cls, property_name)
+                moved_attrs.add(property_name)
 
 
 class ABCCachedProperties(CachedProperties, ABCMeta):

--- a/homeassistant/helpers/entity.py
+++ b/homeassistant/helpers/entity.py
@@ -339,7 +339,7 @@ class CachedProperties(type):
             moved_attrs.add(property_name)
 
         for parent in cls.__mro__[:0:-1]:
-            if not hasattr(parent, "_CachedProperties__cached_properties"):
+            if "_CachedProperties__cached_properties" not in parent.__dict__:
                 continue
             cached_properties = getattr(parent, "_CachedProperties__cached_properties")
             for property_name in cached_properties:

--- a/homeassistant/helpers/entity.py
+++ b/homeassistant/helpers/entity.py
@@ -328,7 +328,7 @@ class CachedProperties(type):
                 annotations = cls.__annotations__
                 if attr_name in annotations:
                     annotations[private_attr_name] = annotations.pop(attr_name)
-            type.__setattr__(cls, attr_name, make_property(property_name))
+            setattr(cls, attr_name, make_property(property_name))
 
         cached_properties: set[str] = namespace["_CachedProperties__cached_properties"]
         moved_attrs: set[str] = set()

